qiskit/circuit/gate.py

```diff
@@ -40,6 +40,16 @@ def __init__(
     # Set higher priority than Numpy array and matrix classes
     __array_priority__ = 20
 
+    def __repr__(self):
+        """Generates a representation of the Gate object instance
+        Returns:
+            str: A representation of the Gate instance with class name, name,
+                 number of qubits, number of classical bits and  params( if any )"""
+        return (
+            f"<{self.__class__.__name__} '{self.name}' labeled '{self.label}' with"
+            f" {self.num_qubits} qubits, {self.num_clbits} clbits and params={self.params}>"
```

Comment:
Could we only include the "labeled" bit if `self.label is not None`?  Otherwise we'll end up with the slightly confusing behaviour of this reporting that something is `labeled 'None'` in most cases.   (Same comment on all other `label` parts.)